src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -405,29 +417,77 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 								 quotedShardName);
 			}
 
+			int32 shardPlacementGroupId = shardPlacement->groupId;
+
 			/*
 			 * The active DROP SCHEMA/DATABASE ... CASCADE will drop the shard, if we
 			 * try to drop it over another connection, we will get into a distributed
 			 * deadlock.
 			 */
-			if (shardPlacement->groupId == COORDINATOR_GROUP_ID &&
+			if (shardPlacementGroupId == COORDINATOR_GROUP_ID &&
 				IsCoordinator() &&
 				DropSchemaOrDBInProgress())
 			{
 				DeleteShardPlacementRow(shardPlacement->placementId);
 				continue;
 			}
 
+			/*
+			 * If it is a local placement of a distributed table, then execute
+			 * the DROP command locally.
+			 */
+			if (shardPlacementGroupId == localGroupId)
```

Comment:
We should create the `DROP` tasklist at once. Otherwise, we will use local execution needlessly for any `DROP` command on regular dist. table